k8s/examples/jenkins/20-statefulset.yaml

```diff
@@ -0,0 +1,153 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: jenkins-master
+  namespace: default
+  labels:
+    app: jenkins
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: jenkins
+      component: jenkins-master
+      storageos.com/replicas: "1"
+      storageos.com/fenced: "true"
+  serviceName: jenkins
+  template:
+    metadata:
+      labels:
+        app: jenkins
+        component: jenkins-master
+        storageos.com/replicas: "1"
+        storageos.com/fenced: "true"
+    spec:
+      securityContext:
+        runAsUser: 1000
+        fsGroup: 1000
+      serviceAccountName: "jenkins"
+      initContainers:
+        - name: "copy-default-config"
+          image: "jenkins/jenkins:lts"
+          imagePullPolicy: "Always"
+          command: [ "sh", "/var/jenkins_config/apply_config.sh" ]
+          env:
+            - name: ADMIN_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: jenkins
+                  key: jenkins-admin-password
+            - name: ADMIN_USER
+              valueFrom:
+                secretKeyRef:
+                  name: jenkins
+                  key: jenkins-admin-user
+          resources:
+            limits:
+              cpu: 2000m
+              memory: 4096Mi
+            requests:
+              cpu: 50m
+              memory: 256Mi
+          volumeMounts:
+            - mountPath: /tmp
+              name: tmp
+            - mountPath: /var/jenkins_home
+              name: jenkins-home
+            - mountPath: /var/jenkins_config
+              name: jenkins-config
+            - mountPath: /usr/share/jenkins/ref/secrets/
+              name: secrets-dir
+            - mountPath: /usr/share/jenkins/ref/plugins
+              name: plugins
+            - mountPath: /var/jenkins_plugins
+              name: plugin-dir
+      containers:
+        - name: jenkins
+          image: "jenkins/jenkins:lts"
+          imagePullPolicy: "Always"
+          args: [ "--argumentsRealm.passwd.$(ADMIN_USER)=$(ADMIN_PASSWORD)",  "--argumentsRealm.roles.$(ADMIN_USER)=admin"]
+          env:
+            - name: JAVA_OPTS
```

Comment:
You can add the JVM memory specs here and disable the SetupWizard at start (since you're injecting the plugins by file)

For instance
```
       env:
        - name: JAVA_OPTS
          value: '-Xmx1400m -Djenkins.install.runSetupWizard=false'
# arbitrary mem number
```

Reply:
There's no need to disable the Setup Wizzard as there's config being loaded via a configmap so setup doesn't run anyway.

Reply:
Set -Xmx1500m